src/view/states/State.h

```diff
@@ -6,19 +6,40 @@
 #define SIMPLETASKMANAGER_STATE_H
 
 #include "user_interface/StateContext.h"
-#include "user_interface/console_io/ConsolePrinter.h"
-#include "user_interface/console_io/ConsoleReader.h"
 
 #include <memory>
 #include <optional>
 
+enum class StateType;
+
 class State {
 public:
-    virtual std::shared_ptr<State> Execute(StateContext&) = 0;
+    virtual StateType Execute(StateContext&) = 0;
 
 public:
     virtual ~State() = default;
 };
 
+enum class StateType {
```

Comment:
StateType feels more like implementation, not interface. Looks strange to bring concrete steps names to Step.h, but I don't have any suggestions at the moment